### ISSUE TITLE
Padding="same" documentation is incorrect in all Conv layers

### ISSUE THREAD

**Issue by kkraoj** [keras-team/keras#15703]
**URL(s) with the issue:**.
https://keras.io/api/layers/convolution_layers/convolution1d/
https://keras.io/api/layers/convolution_layers/convolution2d/
https://keras.io/api/layers/convolution_layers/convolution3d/
https://keras.io/api/layers/convolution_layers/separable_convolution1d/
https://keras.io/api/layers/convolution_layers/separable_convolution2d/
https://keras.io/api/layers/convolution_layers/depthwise_convolution2d/
https://keras.io/api/layers/convolution_layers/convolution2d_transpose/
https://keras.io/api/layers/convolution_layers/convolution3d_transpose/

**Description of issue (what needs to be changed):**.
The documentation for padding in all conv layers listed above states that, " "same" results in padding with zeros evenly to the left/right or up/down of the input such that **output has the same height/width dimension as the input.**". The part in bold is incorrect (example shown below). setting padding = "same" only means that enough padding is applied to the input so that the kernel can fully cover the input.

Example:
```
from tensorflow import keras
from tensorflow.keras import layers

def get_model(img_size, num_classes):
    inputs = keras.Input(shape=img_size + (3,))
    x = layers.Conv2D(32, 3, strides=2, padding="same")(inputs)
    outputs = layers.Conv2D(64, 3, strides=1, padding="same")(x)

    # Define the model
    model = keras.Model(inputs, outputs)
    return model

img_size = (224, 224)
num_classes = 2
model = get_model(img_size, num_classes)
model.summary()
```
Results in:
```
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
=================================================================
 input_4 (InputLayer)        [(None, 224, 224, 3)]     0         
                                                                 
 conv2d_4 (Conv2D)           (None, 112, 112, 32)      896       <-**Padding=same did not preserve image shape**
                                                                 
 conv2d_5 (Conv2D)           (None, 112, 112, 64)      18496     <-**Padding=same preserved image shape only because strides=1**
                                                                 
=================================================================
```
As you can see, when `strides=2`, the image shape is not preserved, it is halved in spite of setting `padding="same"`. Only when `strides=1`, the input shape is preserved. Thus the current documentation `padding` reflects its behaviour only in the special case of `strides=1` and not in the general case. I suggest two solutions: 

1. Reword the documentation to: "same" results in padding with zeros evenly to the left/right or up/down of the input such that **the kernel fits is able to see the whole input**".
2. or reword the documentation to:  "same" results in padding with zeros evenly to the left/right or up/down of the input such that output has the same height/width dimension as the input **when the stride is 1**."

**Correct links**.
Is the link to the source code correct?
Yes

**Parameters defined**.

Are all parameters defined and formatted correctly?
Yes

**Returns defined**.

Are return values defined?
Yes

**Raises listed and defined**

Are the errors defined?
Yes

**Usage example**

Is there a usage example?
See the API guide: https://www.tensorflow.org/community/contribute/docs_ref on how to write testable usage examples.
Yes

**Submit a pull request?**.

Are you planning to also submit a pull request to fix the issue? See the [docs contributor guide](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md):

Yes

**Comment by sanatmpa1** [comment]
@kkraoj,

As you pointed out, I can see that the `padding=same` preserves the input dimensions only when `stride=1`, and I can submit a PR to modify the documentation as 

> "same" results in padding with zeros evenly to the left/right or up/down of the input such that the kernel can fully cover the input"

Please confirm if that looks good, Thanks!

**Comment by kkraoj** [comment]
I confirm. That change looks good to me.

**Comment by sanatmpa1** [comment]
@kkraoj,

I have submitted a [PR](https://github.com/keras-team/keras/pull/15771) addressing the doc string change. Thanks!

**Comment by tilakrayal** [comment]
Hello, Thank you for reporting an issue.

We're currently in the process of migrating the new `Keras 3` code base from **keras-team/keras-core** to **keras-team/keras.**
Consequently, This issue may not be relevant to the `Keras 3 `code base. After the migration is successfully completed, feel free to reopen this issue at **keras-team/keras** if you believe it remains relevant to the `Keras 3` code base. 
If instead this issue is a bug or security issue in legacy **tf.keras**, you can instead report a new issue at [keras-team/tf-keras](https://github.com/keras-team/tf-keras/issues), which hosts the TensorFlow-only, legacy version of Keras.

To know more about `Keras 3,` please take a look at https://keras.io/keras_core/announcement/. Thank you!
